README.md

```diff
@@ -127,6 +127,45 @@ a working system.
 
 **Note:** If every role does not go green in `pod-status --watch` refer to [Troubleshooting](#troubleshooting)
 
+3. Pulling updates
+
+    When you want to pull the latest changes from the upstream you should:
+
+    ```
+    # Pull the changes (or checkout the commit you want):
+    git pull
+
+    # Update all submodules to match the checked out commit
+    git submodules update --recursive
+    ```
+
+    Sometimes, when we bump the BOSH release submodules, they move to a different
+    location and you need to run:
+
+    ```
+      git submodule sync --recursive
+    ```
+
+    You might have to run the `git submodules update --recursive` again after the
+    last command.
+
+    If there are untracked changes from submodule directories you can safely remove them.
+
+    E.g. A command that will update all submodules and drop any changed or untracked files in them is:
```

Comment:
I added this example because sometimes when changing branches it is really hard to get rid of the untracked files and everything. I think the warning is also necessary here.